File: suraj_24b0907/week7/fibonacci.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> v={0,1};
    int fib(int n) {
        if(n<v.size()) return v[n];
        while (v.size() <= n){
            v.push_back(v[v.size() - 1] + v[v.size() - 2]);}
        return v[n];
    }
};

File: Chavda_23B0368/Week_7/Fibonacci.java
package Week_7;

class Solution {
    public int fib(int n) {
        if (n == 0) return 0;
        if (n == 1) return 1;
        int prev = 1, prev2 = 0;
        int ans = 0;
        for (int i = 1; i < n; i++) {
            ans = prev + prev2;
            prev2 = prev;
            prev = ans;
        }
        return ans;
    }
}